js/console.go

```diff
@@ -22,20 +22,42 @@ package js
 
 import (
 	"context"
+	"os"
 	"strconv"
 
 	"github.com/dop251/goja"
 	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
 )
 
 type Console struct {
 	Logger *log.Logger
 }
 
+// NewConsole creates a console with the standard logrus logger.
 func NewConsole() *Console {
 	return &Console{log.StandardLogger()}
 }
 
+// NewFileConsole creates a console logger with its output set to the file at the provided `filepath`.
+func NewFileConsole(fs afero.Fs, filepath string) (*Console, error) {
+	f, err := fs.OpenFile(filepath, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
+	if err != nil {
+		if os.IsNotExist(err) {
+			if f, err = fs.Create(filepath); err != nil {
```

Comment:
So, this may take some time to resolve. `os.O_CREATE` doesn't work with `afero.CacheOnReadFs` that we use as the base FS for the runner. I have raised issue spf13/afero#193 to track this.